### DISCUSSION TITLE
How do I mutate data without using a form?

### DISCUSSION THREAD

**Discussion by juniorforlife** [remix-run/remix#5864] [Q&A] [answered]
All the examples and tutorials about mutation in Remix is showing how to do it with a form. How do I send a POST/PUT/DELETE request without a form? An example would be something like Tinder, you send a request to update data when user swipes left or right. No form is needed in this case.

**Comment by sergiodxa** [comment, marked as answer]
You can use the [useSubmit](https://remix.run/docs/en/1.14.3/hooks/use-submit) or [useFetcher](https://remix.run/docs/en/1.14.3/hooks/use-fetcher) hooks too.

**Reply by juniorforlife** [reply]
ah.. I did see this page about useSubmit but the confusing part is
```
export async function action({ request }: ActionArgs) {
  await updatePreferences(await request.formData());
  return redirect("/prefs");
}
```
so let's say I want to submit with a product id then that id will be stored in ` request.formData()` right?

**Reply by sergiodxa** [reply]
Yes, Remix always use FormData, even if you don’t use a form tag or the Form component.
